tests/commands/test_cmd_deploy.py

```diff
@@ -84,3 +84,31 @@ def test_multiple_templates(mock_deployment_wait, runner, json_fixture):
     result = runner(['deploy', '--template', 'tests/test.json.tmpl', '--template', 'tests/test-2.json.tmpl'], env=env)
 
     assert result.exit_code == 0
+
+
+@responses.activate
+def test_dry_run(runner, json_fixture):
+    env = {
+        'SHPKPR_MARATHON_URL': "http://marathon.somedomain.com:8080",
+        'SHPKPR_APPLICATION': 'test-app',
+        'SHPKPR_DOCKER_REPOTAG': 'goexample/outyet:latest',
+        'SHPKPR_DOCKER_EXPOSED_PORT': '8080',
+        'SHPKPR_DEPLOY_DOMAIN': 'mydomain.com',
+    }
+    result = runner(['deploy', '--dry-run', '--template', 'tests/test.json.tmpl', 'RANDOM_LABEL=some_value'], env=env)
+
+    assert result.exit_code == 0
+    assert '--dry-run' in result.output
+
+
+@responses.activate
+def test_dry_run_fail(runner, json_fixture):
```

Comment:
Yeah, the test is a bit ambiguous, will clean it up and try to clarify. Basically this is testing that given *some random pre-deploy failure* (it really doesn't matter what, tbh) the script still exits with a non-zero exit code when `--dry-run` is enabled.